CIPs/0008.md

```diff
@@ -0,0 +1,74 @@
+# CIP 0008: Expand Metadata to general off-chain storage
+
+- Date: 2020-05-05
+- Author: @nambrot
+- Status: WIP
+
+## Overview
+
+In CIP3, the Celo protocol has started with a simple off-chain storage mechanism mainly focused on claims an account can make about their off-chain identity and communicate to other participants in the network. Over time, there have been additional use cases which have similar access patterns that prompted the creation of `Metadata` in the first place. I.e. data that does not require the expensive cost of "hard consensus", instead just needs to be authenticated by a single party.
```

Comment:
Is there reference code to verify the CIP3 signature anywhere?
Currently it says to concatenate and hash, but not sure if it pays attention to whitespace, encoding, etc.

Reply:
Yeah it's not super well specified, but here is the code CK currently uses https://github.com/celo-org/celo-monorepo/blob/master/packages/contractkit/src/identity/claims/claim.ts#L89